packages/account/src/Sections/Profile/PhoneVerification/__test__/confirm-phone-number.spec.tsx

```diff
@@ -0,0 +1,157 @@
+import React from 'react';
+import { act, render, screen } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import { StoreProvider, mockStore } from '@deriv/stores';
+import { usePhoneNumberVerificationSetTimer, useRequestPhoneNumberOTP, useSettings } from '@deriv/hooks';
+import ConfirmPhoneNumber from '../confirm-phone-number';
+
+jest.mock('@deriv/hooks', () => ({
+    ...jest.requireActual('@deriv/hooks'),
+    useRequestPhoneNumberOTP: jest.fn(() => ({
+        error_message: '',
+        requestOnWhatsApp: jest.fn(),
+        requestOnSMS: jest.fn(),
+        setErrorMessage: jest.fn(),
+        setUsersPhoneNumber: jest.fn(),
+    })),
+    useSettings: jest.fn(() => ({
+        data: {},
+        invalidate: jest.fn(),
+    })),
+    usePhoneNumberVerificationSetTimer: jest.fn(() => ({
+        next_otp_request: undefined,
+    })),
+}));
+
+describe('ConfirmPhoneNumber', () => {
+    const store = mockStore({
+        ui: {
+            setShouldShowPhoneNumberOTP: jest.fn(),
+        },
+    });
+
+    const mockSetOtp = jest.fn();
+    const whatsapp_button_text = 'Get code via WhatsApp';
+    const sms_button_text = 'Get code via SMS';
+
+    it('should render ConfirmPhoneNumber', () => {
+        (useSettings as jest.Mock).mockReturnValue({
+            data: { phone: '+0123456789' },
+        });
+        render(
+            <StoreProvider store={store}>
+                <ConfirmPhoneNumber setOtpVerification={mockSetOtp} />
+            </StoreProvider>
+        );
+        const phone_number_textfield = screen.getByRole('textbox', { name: 'Phone number' });
+        expect(screen.getByText('Confirm your phone number')).toBeInTheDocument();
+        expect(phone_number_textfield).toBeInTheDocument();
+        expect(phone_number_textfield).toHaveValue('0123456789');
+        expect(screen.getByRole('button', { name: sms_button_text })).toBeInTheDocument();
+        expect(screen.getByRole('button', { name: whatsapp_button_text })).toBeInTheDocument();
+    });
+
+    it('should call setErrorMessage when the user presses a key', async () => {
+        const mock_set_error_message = jest.fn();
+        (useRequestPhoneNumberOTP as jest.Mock).mockReturnValue({
+            setErrorMessage: mock_set_error_message,
+        });
+        render(
+            <StoreProvider store={store}>
+                <ConfirmPhoneNumber setOtpVerification={mockSetOtp} />
+            </StoreProvider>
+        );
+        const phone_number_textfield = screen.getByRole('textbox', { name: 'Phone number' });
+        expect(screen.getByText('Confirm your phone number')).toBeInTheDocument();
+        expect(phone_number_textfield).toBeInTheDocument();
+        userEvent.clear(phone_number_textfield);
+        await act(async () => userEvent.type(phone_number_textfield, '+01293291291'));
+        expect(mock_set_error_message).toHaveBeenCalled();
```

Comment:
the message to wrap in act in console is misleading.
userEvent is already wrapped in the act,
may I sk you you try to wrap expectation to waitFor instead 🙏 
```suggestion
        userEvent.type(phone_number_textfield, '+01293291291');
       await waitFor( ()=>{
       expect(mock_set_error_message).toHaveBeenCalled();
       })
```